From a3a9faac9cd81b84d6ec1d3f87651d65e6c1c204 Mon Sep 17 00:00:00 2001
From: ltian <le.tian@windriver.com>
Date: Mon, 15 Nov 2010 10:31:33 +0800
Subject: [PATCH 061/132] net: Fix compilation errors.

Fix printk format string problem. Change use of __ilog2 (which is only
available on ppc) to __ilog2_u32.

Signed-off-by: Chandrakala Chavva <cchavva@caviumnetworks.com>
Signed-off-by: ltian <le.tian@windriver.com>
---
 drivers/net/rionet.c |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/net/rionet.c b/drivers/net/rionet.c
index d3cd7f8..b32001a 100644
--- a/drivers/net/rionet.c
+++ b/drivers/net/rionet.c
@@ -304,8 +304,8 @@ static int rionet_queue_tx_msg(struct sk_buff *skb, struct net_device *ndev,
 		netif_wake_queue(ndev);
 #endif
 	if (netif_msg_tx_queued(rnet))
-		printk(KERN_INFO "%s: queued skb %8.8x len %8.8x\n", DRV_NAME,
-		       (u32) skb, skb->len);
+		printk(KERN_INFO "%s: queued skb %p len %8.8x\n", DRV_NAME,
+		       skb, skb->len);
 
 	return 0;
 }
@@ -747,7 +747,7 @@ static void rionet_remove(struct rio_dev *rdev)
 	struct rionet_peer *peer, *tmp;
 
 	free_pages((unsigned long)rionet_active, rdev->net->hport->sys_size ?
-					__ilog2(sizeof(void *)) + 4 : 0);
+					__ilog2_u32(sizeof(void *)) + 4 : 0);
 	unregister_netdev(ndev);
 	kfree(ndev);
 
@@ -815,7 +815,7 @@ static int rionet_setup_netdev(struct rio_mport *mport)
 	}
 
 	rionet_active = (struct rio_dev **)__get_free_pages(GFP_KERNEL,
-			mport->sys_size ? __ilog2(sizeof(void *)) + 4 : 0);
+			mport->sys_size ? __ilog2_u32(sizeof(void *)) + 4 : 0);
 	if (!rionet_active) {
 		rc = -ENOMEM;
 		goto out;
-- 
1.6.5.2

